util/fibers/fibers_test.cc

```diff
@@ -616,5 +614,75 @@ TEST_P(ProactorTest, Mutex) {
     th.reset();
 }
 
+TEST_P(ProactorTest, DragonflyBug1591) {
+  unique_ptr<ProactorThread> ths[kNumThreads];
+  Fiber fbs[kNumThreads];
+
+  auto sock = std::unique_ptr<FiberSocketBase>(proactor()->CreateSocket());
+  auto sock2 = std::unique_ptr<FiberSocketBase>(proactor()->CreateSocket());
```

Comment:
wow, seems it was not easy to reproduce even when you knew exactly what happens.